controllers/reconcile.go

```diff
@@ -4,45 +4,73 @@ import (
 	"context"
 
 	rolloutsmanagerv1alpha1 "github.com/argoproj-labs/argo-rollouts-manager/api/v1alpha1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func (r *RolloutManagerReconciler) reconcileRolloutsManager(ctx context.Context, cr *rolloutsmanagerv1alpha1.RolloutManager) error {
+func (r *RolloutManagerReconciler) reconcileRolloutsManager(ctx context.Context, cr *rolloutsmanagerv1alpha1.RolloutManager) (*metav1.Condition, error) {
+
+	log.Info("Searching for existing RolloutManager")
+	if err := checkForExistingRolloutManager(ctx, r.Client, cr, log); err != nil {
+		if doMultipleRolloutManagersExist(err) {
+			return createCondition(rolloutsmanagerv1alpha1.RolloutManagerConditionTypeErrorOccurred, metav1.ConditionTrue, rolloutsmanagerv1alpha1.RolloutManagerReasonMultipleClusterScopedRolloutManager, err.Error()), err
```

Comment:
For this line, I think we can return `nil` rather than error:
`return createCondition(rolloutsmanagerv1alpha1.RolloutManagerConditionTypeErrorOccurred, metav1.ConditionTrue, rolloutsmanagerv1alpha1.RolloutManagerReasonErrorOccurred, err.Error()), nil`

We don't want to return an error in this case, because it is a user error, and thus there is nothing the we can do here. Any error that is returned will just cause the Reconcile() function to be called again and again.

(The line below this one SHOULD return nil, however, that is correct).